astropy/io/fits/hdu/image.py

```diff
@@ -1008,7 +1008,7 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
     def match_header(cls, header):
         card = header.cards[0]
         return (card.keyword == 'SIMPLE' and
-                ('GROUPS' not in header or not header['GROUPS']) and
+                ('GROUPS' not in header or header['GROUPS'] is not True) and
```

Comment:
OK, this sounds sensible. Our test fits cover cases for both ``'T'`` and ``'F'``, but you're right that there may be all kind of other values now.